### PR TITLE
STYLE: In-class default member initializers for ExhaustiveOptimizer data

### PR DESCRIPTION
**Pull request by N-Dekker**
Follow-up to pull request #3102 and pull request #3118, both by Jon Haitz Legarreta Gorroño (@jhlegarreta).

### CODE REVIEW

**Comment by N-Dekker** [comment]
The empty `{}` default member initializers that I added with this pull request are not strictly necessary. They are just there to make it _absolutely clear_ that none of the data member were forgotten, anymore.

Those empty `{}` default member initializers would be really needed for types that do _not_ have a user-defined default-constructor. However, it may not be obvious at a glance whether or not types like `StepsType` and `ParametersType` have a user-defined default-constructor.

File: Modules/Numerics/Optimizers/src/itkExhaustiveOptimizer.cxx [hunk 1]
```diff
@@ -25,16 +25,6 @@ namespace itk
 ExhaustiveOptimizer::ExhaustiveOptimizer()
 {
   itkDebugMacro("Constructor");
```
Comment (blowekamp) [inline Modules/Numerics/Optimizers/src/itkExhaustiveOptimizer.cxx:27]: Is this still useful? Could we just use the default constructor?

Comment (N-Dekker) [reply Modules/Numerics/Optimizers/src/itkExhaustiveOptimizer.cxx:27]: @blowekamp Good question. 😃  This `itkDebugMacro("Constructor")` certainly isn't useful to me. I do see them in a few other places in ITK, though. Shall we remove them all?

If we would remove the `itkDebugMacro("Constructor")` call, we could certainly do `ExhaustiveOptimizer::ExhaustiveOptimizer() = default`.

Comment (dzenanz) [reply Modules/Numerics/Optimizers/src/itkExhaustiveOptimizer.cxx:27]: These sort of debug statements are useful during initial development, and occasionally during some later debugging. As this is quite a mature class by now, I vote we remove it and go with `= default;`.

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**